app/src/main/java/com/github/libretube/ui/preferences/BackupRestoreSettings.kt

```diff
@@ -77,18 +85,56 @@ class BackupRestoreSettings : BasePreferenceFragment() {
         }
     }
 
+    private fun createDialog(
+        titleStringId: Int,
+        items: List<String>,
+        onConfirm: (Int) -> Unit
+    ) {
+        var selectedIndex = 0
+        MaterialAlertDialogBuilder(this.requireContext())
+            .setTitle(getString(titleStringId))
+            .setSingleChoiceItems(items.toTypedArray(), selectedIndex) { _, i ->
+                selectedIndex = i
+            }
+            .setPositiveButton(
+                R.string.okay
+            ) { _, _ -> onConfirm(selectedIndex) }
+            .setNegativeButton(R.string.cancel, null)
+            .show()
+    }
+
+    private fun importFormatList(): List<String> {
+        return ImportFormat.values().map {
+            when (it) {
+                ImportFormat.NEWPIPE -> getString(R.string.import_format_newpipe)
+                ImportFormat.FREETUBE -> getString(R.string.import_format_freetube)
+                ImportFormat.YOUTUBECSV -> getString(R.string.import_format_youtube_csv)
+            }
+        }
+    }
+
+    private fun exportFormatList(): List<String> {
+        return listOf(getString(R.string.import_format_newpipe), getString(R.string.import_format_freetube))
+    }
```

Comment:
These functions should be removed in favor of the approach of having the string resource directly inside the enum (see my suggestion in the `SupportedClient.kt` file (which should be renamed to `ImportFormat.kt` by the way).